src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -114,6 +119,11 @@ public class MultiSyncModeSelector : ISyncModeSelector, IDisposable
             _pivotNumber = _syncConfig.PivotNumberParsed;
             _isSnapSyncDisabledAfterAnyStateSync = _syncProgressResolver.FindBestFullState() != 0;
 
+            Changed += (src, args) =>
+            {
+                ReportSink.CurrentStage = args.Current;
+            };
+
```

Comment:
Don't like it, it makes the dependencies go the other way around. Also static class makes it unparallelizable.